opencti-platform/opencti-front/src/private/components/settings/users/UserConfidenceLevel.tsx

```diff
@@ -0,0 +1,81 @@
+import React from 'react';
+import { User_user$data } from '@components/settings/users/__generated__/User_user.graphql';
+import { Link } from 'react-router-dom-v5-compat';
+import { ReportGmailerrorred } from '@mui/icons-material';
+import Tooltip from '@mui/material/Tooltip';
+import { InformationOutline } from 'mdi-material-ui';
+import Box from '@mui/material/Box';
+import { useFormatter } from '../../../../components/i18n';
+
+type Data_UserConfidenceLevel = User_user$data['user_confidence_level'];
+type Data_EffectiveConfidenceLevel = User_user$data['effective_confidence_level'];
+
+type UserConfidenceLevelProps = {
+  confidenceLevel?: Data_UserConfidenceLevel | Data_EffectiveConfidenceLevel
+};
+
+const UserConfidenceLevel: React.FC<UserConfidenceLevelProps> = ({ confidenceLevel }) => {
+  const { t_i18n } = useFormatter();
+
+  if (!confidenceLevel) {
+    return (
+      <Tooltip
+        title={t_i18n("No confidence level found in this user's groups, and no confidence level defined at the user level.")}
+      >
+        <ReportGmailerrorred fontSize={'small'} color={'error'}/>
+      </Tooltip>
+    );
+  }
+
+  // TODO: add overrides in a tooltip when in use
+
+  const renderSource = () => {
+    const source = (confidenceLevel as Data_EffectiveConfidenceLevel)?.source;
+
+    // FIXME: if watching the current user's detailed view, the source is {}, hence the check if (source.entity_type && ...) below
```

Comment:
FIXME later ?

Reply:
I did not have time to investigate, but it's probably related to how we poorly handle the graphql cache records

Warning: RelayModernRecord: Invalid record update, expected both versions of record `88ec0c6a-13ce-5e39-b486-354fe4a7084f` to have the same `__typename` but got conflicting types `MeUser` and `User`. The GraphQL server likely violated the globally unique id requirement by returning the same id for different objects.

Another whole issue to address, certainly not in this PR.

Reply:
I'm editing myself, which collides with `MeUser` query for instance. The record gets "updated" wrongly somehow.